core/src/main/scala/chisel3/properties/Class.scala

```diff
@@ -165,11 +173,53 @@ object Class {
     // Bind the Property[ClassType] type for this Object.
     contextMod match {
       case rm: RawModule => {
-        rm.addCommand(DefObject(sourceInfo, obj))
+        rm.addCommand(DefObject(sourceInfo, obj, obj.className.name))
+        classProp.bind(OpBinding(rm, Builder.currentWhen), SpecifiedDirection.Unspecified)
+      }
+      case cls: Class => {
+        cls.addCommand(DefObject(sourceInfo, obj, obj.className.name))
+        classProp.bind(ClassBinding(cls), SpecifiedDirection.Unspecified)
+      }
+      case _ => throwException("Internal Error! Property connection can only occur within RawModule or Class.")
+    }
+
+    obj
+  }
+
+  def getReferenceType(cls: Definition[Class]): Property[ClassType] = {
+    // Get the BaseModule for the Class this is a definition of.
+    val baseModule = cls.getInnerDataContext match {
+      case Some(b) => b
+      case _       => throwException("Internal Error! Class instance did not have an associated BaseModule.")
```

Comment:
```suggestion
    val baseModule = cls.getInnerDataContext.getOrElse(
      throwException("Internal Error! Class instance did not have an associated BaseModule.")
    )
```
Nitty suggestion but you could use `.getOrElse` here.